python/dolfinx/fem/petsc.py

```diff
@@ -1118,3 +1119,61 @@ def interpolation_matrix(space0: _FunctionSpace, space1: _FunctionSpace) -> PETS
         Interpolation matrix.
     """
     return _interpolation_matrix(space0._cpp_object, space1._cpp_object)
+
+
+@functools.singledispatch
+def assign(u: typing.Union[_Function, Sequence[_Function]], x: PETSc.Vec):
```

Comment:
Is the reasoning for keeping this assign in `fem.petsc` that it works on a `Function` rather than a set of arrays?

Reply:
Yes. 

The linear algebra module should not see/include any finite element concepts.